roles/test_operator/tasks/stages.yml

```diff
@@ -0,0 +1,45 @@
+---
+- name: "Run stage {{ _stage_vars.name }}"
+  ansible.builtin.debug:
+    msg: "Run stage {{ _stage_vars.name }}"
+
+- name: "Call pre stage hooks for test operator stage {{ _stage_vars.name }}"
+  when:
+    - not cifmw_test_operator_dry_run | bool
+    - _stage_vars.pre_test_stage_hooks is defined
+  vars:
+    step: pre_test_hooks
+    hooks: "{{ _stage_vars.pre_test_stage_hooks }}"
+  ansible.builtin.import_role:
+    name: run_hook
+
+- name: Include stage var file
+  ansible.builtin.include_vars:
+    file: "{{ _stage_vars.test_vars_file | default('/dev/null') }}"
+    name: _stage_var_file
+
+- name: Set stage var fact
+  vars:
+    file_vars: "{{ _stage_var_file }}"
+    test_vars: "{{ _stage_vars.test_vars | default({}) }}"
+  ansible.builtin.set_fact:
+    _stage_test_vars: "{{ file_vars | combine(test_vars) }}"
+
+- name: Overwrite global_vars with stage_vars
+  with_dict: "{{ vars }}"
+  when: item.key.startswith('cifmw_test_operator')
+  ansible.builtin.set_fact:
+    "{{ item.key | replace('cifmw', '') }}": "{{ _stage_test_vars[item.key] | default(item.value) | default(omit) }}"
+
+- name: "Call runner {{ _stage_vars.type }}"
+  ansible.builtin.include_tasks: "runners/{{ _stage_vars.type }}_runner.yml"
+
```

Comment:
Hi Lukas I was looking into the undefine we wanted to add the the role using undef() but after looking into it even more it won't work having that it can only override variables of lesser precedence
```
The Jinja2 undef() function returns a Python AnsibleUndefined object, derived from jinja2.StrictUndefined. Use undef() to undefine variables of lesser precedence. For example, a host variable can be overridden for a block of tasks:
```
https://docs.ansible.com/ansible/latest/playbook_guide/playbooks_templating_undef.html
[6:37](https://redhat-internal.slack.com/archives/D073LFRFD36/p1728920250531589)
I did add something that should allow us to be safe without undefining the variables here https://github.com/openstack-k8s-operators/ci-framework/pull/2374/files#diff-e4102a0384414f2b689daca5a01621d2a7af8ff3873200e113a3b1c603525f48R29

hope it is enough to cover that front

Reply:
Let's talk this through during the meeting. I think that this issue together with this one [1] are the two major bottlenecks of this PR.

[1] https://github.com/openstack-k8s-operators/ci-framework/pull/2374/files#r1795170859